step-release-vis/src/app/components/polygon/polygon.ts

```diff
@@ -0,0 +1,24 @@
+import {Component, Input, OnInit} from '@angular/core';
+import {Polygon} from '../../models/Polygon';
+
+@Component({
+  selector: 'app-polygon',
+  template: '<svg:polygon [attr.points]="getPoints()"/>',
```

Comment:
I tried checking if having a polygon as a separate component will work, but didn't have much success. Looks like since its a separate component, it wraps the template contents in a component tag, while in SVG we need the polygon tag to be directly nested. I did some research, trying to find solutions on dynamic SVG components, but didn't find anything, that would work, so we'll need help with this part.

Reply:
can we configure what the wrapping selector is? can we make  it match what e want?

Reply:
There is a way of using a selector as an attribute, rather than an element, but I didn't manage to use it with *ngFor yet.

Reply:
Having this as the template for the Environment component works, but then the Polygon component is not used.
```
<svg xmlns="http://www.w3.org/2000/svg" height="1000" width="1000">
  <svg:polygon app-polygon *ngFor="let polygon of polygons" [attr.points]="getPoints(polygon)"></svg:polygon>
</svg>

```

Reply:
On the other hand, maybe the Polygon component isn't really needed, as there is almost no logic in it. It just has a field specifying the polygon points, which is still an attribute of the HTML tag.
We will have a Candidate component later, which knows about all its polygons in different envs.